src/infra/ChatAnalyticsReporter.ts

```diff
@@ -47,9 +51,16 @@ export class ChatAnalyticsReporter {
       Authorization: `KEY ${this.apiKey}`,
       'Content-Type': 'application/json',
     };
+
+    const sessionId: string | undefined = this.sessionTrackingEnabled
+      ? (event.sessionId ?? acquireSessionId() ?? undefined)
+      : undefined;
     const res = await this.httpRequesterService.post(
       this.endpoint,
-      event,
+      {
+        ...event,
+        sessionId
```

Comment:
Should the session ID come before the event, so that you could override it in an individual call if you wanted? Or am I misremembering how spread syntax works?

Reply:
yup, sessionId that is passed by user is handled here: https://github.com/yext/analytics/pull/42/files/a366252031511ca8b62bcaae5bdddfa583dad4a4#diff-cdc1ccc07d263bfd62a8e63ea5692787b1220210e183901c69e8fb84ba7b0fdaR55-R57
provided sessionTrackingEnabled is set to true.

Reply:
Perfect 👍